Azure repository creation Issue in Elasticsearch 2.4.0

Hi All,

I want to take snapshot and restore using azure repository in Linux, I have configured these below setting in yml,

```
##################### Azure cloud plugin for repository ######################

cloud:
  azure:
    storage:
      account: "https://XXX.blob.core.windows.net/"
      key: "..."
```

with the above setting, service is started, but i am m getting below error when creating repository in Elasticsearch 2.4.0,


    PUT snapshot/mybackup
    {
    "type": "azure"
    }

Error msg:-

```json
{
   "error": {
      "root_cause": [
         {
            "type": "repository_exception",
            "reason": "[elasticsearch-snapshots] Can not find an azure client for account [null]"
         }
      ],
      "type": "repository_exception",
      "reason": "[elasticsearch-snapshots] Can not find an azure client for account [null]"
   },
   "status": 500
}
```

When I looked into the ES logs I could see cloud storage account not getting created at the start of elasticsearch

```
2016-09-22 07:30:23,221][INFO ][http                     ] [mue2rhegrdd002_RD_NDX_MASTER] publish_address {10.247.0.4:9200}, bound_addresses {10.247.0.4:9200}
[2016-09-22 07:30:23,221][INFO ][node                     ] [mue2rhegrdd002_RD_NDX_MASTER] started
[2016-09-22 07:30:23,258][INFO ][cluster.service          ] [mue2rhegrdd002_RD_NDX_MASTER] new_master {mue2rhegrdd002_RD_NDX_MASTER}{-mRr-uLKQkyEJwfqLLI_2Q}{10.247.0.4}{10.247.0.4:9300}{data=false, max_local_storage_nodes=1, master=true}, added {{mue2rhegrdd001_RD_NDX_DATA}{O-JSVOlMQXCuwb5nf5XheQ}{10.247.0.5}{10.247.0.5:9260}{max_local_storage_nodes=1, master=false},{mue2rhegrdd001_RD_NDX_MASTER}{DMAPO7O6QEuXmcsP6aGuAQ}{10.247.0.5}{10.247.0.5:9300}{data=false, max_local_storage_nodes=1, master=true},}, reason: zen-disco-join(elected_as_master, [1] joins received)
**[2016-09-22 07:30:24,576][ERROR][cloud.azure.storage      ] [mue2rhegrdd002_RD_NDX_MASTER] can not create azure storage client: Primary and secondary location URIs in a StorageUri must point to the same resource.**
[2016-09-22 07:30:24,589][INFO ][watcher                  ] [mue2rhegrdd002_RD_NDX_MASTER] starting watch service...
[2016-09-22 07:30:24,590][INFO ][watcher                  ] [mue2rhegrdd002_RD_NDX_MASTER] watch service has started
[2016-09-22 07:30:24,631][INFO ][license.plugin.core      ] [mue2rhegrdd002_RD_NDX_MASTER] license [cf8a2152-4f41-4739-8459-4137cac8105b] - valid
[2016-09-22 07:30:25,489][INFO ][gateway                  ] [mue2rhegrdd002_RD_NDX_MASTER] recovered [20] indices into cluster_state
[2016-09-22 07:30:26,021][INFO ][http                     ] [mue2rhegrdd002_RD_NDX_DATA] publish_address {10.247.0.4:9240}, bound_addresses {10.247.0.4:9240}
[2016-09-22 07:30:26,021][INFO ][node                     ] [mue2rhegrdd002_RD_NDX_DATA] started
[2016-09-22 07:30:26,622][INFO ][cluster.service          ] [mue2rhegrdd002_RD_NDX_MASTER] added {{mue2rhegrdd002_RD_NDX_DATA}{eE1n1iLMQom6VEgYfBwJ9g}{10.247.0.4}{10.247.0.4:9260}{max_local_storage_nodes=1, master=false},}, reason: zen-disco-join(join from node[{mue2rhegrdd002_RD_NDX_DATA}{eE1n1iLMQom6VEgYfBwJ9g}{10.247.0.4}{10.247.0.4:9260}{max_local_storage_nodes=1, master=false}])
[2016-09-22 07:30:26,678][INFO ][cluster.service          ] [mue2rhegrdd002_RD_NDX_DATA] detected_master {mue2rhegrdd002_RD_NDX_MASTER}{-mRr-uLKQkyEJwfqLLI_2Q}{10.247.0.4}{10.247.0.4:9300}{data=false, max_local_storage_nodes=1, master=true}, added {{mue2rhegrdd001_RD_NDX_DATA}{O-JSVOlMQXCuwb5nf5XheQ}{10.247.0.5}{10.247.0.5:9260}{max_local_storage_nodes=1, master=false},{mue2rhegrdd001_RD_NDX_MASTER}{DMAPO7O6QEuXmcsP6aGuAQ}{10.247.0.5}{10.247.0.5:9300}{data=false, max_local_storage_nodes=1, master=true},{mue2rhegrdd002_RD_NDX_MASTER}{-mRr-uLKQkyEJwfqLLI_2Q}{10.247.0.4}{10.247.0.4:9300}{data=false, max_local_storage_nodes=1, master=true},}, reason: zen-disco-receive(from master [{mue2rhegrdd002_RD_NDX_MASTER}{-mRr-uLKQkyEJwfqLLI_2Q}{10.247.0.4}{10.247.0.4:9300}{data=false, max_local_storage_nodes=1, master=true}])
**[2016-09-22 07:30:26,711][ERROR][cloud.azure.storage      ] [mue2rhegrdd002_RD_NDX_DATA] can not create azure storage client: Primary and secondary location URIs in a StorageUri must point to the same resource.**
[2016-09-22 07:30:26,764][INFO ][license.plugin.core      ] [mue2rhegrdd002_RD_NDX_DATA] license [cf8a2152-4f41-4739-8459-4137cac8105b] - valid
```

Please let me know your suggestions to resolve this.

Thanks,
Ganeshbabu R

@dadoonet Can you please double check this again?

WIth the following looks like failing as well

```
cloud:
  azure:
    storage:
        my_account:
            account: "xxxxx...."
            key: "xxxxx...."

```

And getting the following error:

```
[2016-09-26 03:51:43,720][WARN ][rest.suppressed          ] path: /_snapshot/myrepository, params: {repository=myrepository}
RemoteTransportException[[zzz][xxxxx][cluster:admin/repository/put]]; nested: RepositoryException[[elasticsearch-snapshots] Can not find an azure client for account [null]];
Caused by: RepositoryException[[elasticsearch-snapshots] Can not find an azure client for account [null]]
	at org.elasticsearch.cloud.azure.storage.AzureStorageServiceImpl.createContainer(AzureStorageServiceImpl.java:168)
	at org.elasticsearch.cloud.azure.blobstore.AzureBlobStore.createContainer(AzureBlobStore.java:117)
	at org.elasticsearch.repositories.azure.AzureRepository.startVerification(AzureRepository.java:179)
	at org.elasticsearch.repositories.RepositoriesService.verifyRepository(RepositoriesService.java:211)
	at org.elasticsearch.repositories.RepositoriesService$VerifyingRegisterRepositoryListener.onResponse(RepositoriesService.java:436)
	at org.elasticsearch.repositories.RepositoriesService$VerifyingRegisterRepositoryListener.onResponse(RepositoriesService.java:421)
	at org.elasticsearch.cluster.AckedClusterStateUpdateTask.onAllNodesAcked(AckedClusterStateUpdateTask.java:63)
	at org.elasticsearch.cluster.service.InternalClusterService$SafeAckedClusterStateTaskListener.onAllNodesAcked(InternalClusterService.java:733)
	at org.elasticsearch.cluster.service.InternalClusterService$AckCountDownListener.onNodeAck(InternalClusterService.java:1013)
	at org.elasticsearch.cluster.service.InternalClusterService$DelegetingAckListener.onNodeAck(InternalClusterService.java:952)
	at org.elasticsearch.cluster.service.InternalClusterService.runTasksForExecutor(InternalClusterService.java:637)
	at org.elasticsearch.cluster.service.InternalClusterService$UpdateTask.run(InternalClusterService.java:772)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:231)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:194)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

--- a/plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceImpl.java
+++ b/plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceImpl.java
@@ -175,7 +175,7 @@ public class AzureStorageServiceImpl extends AbstractComponent implements AzureS
             blobContainer.createIfNotExists();
         } catch (IllegalArgumentException e) {
             logger.trace((Supplier<?>) () -> new ParameterizedMessage("fails creating container [{}]", container), e);
-            throw new RepositoryException(container, e.getMessage());
+            throw new RepositoryException(container, e.getMessage(), e);
         }
     }
 

